AST.fs

```diff
@@ -240,6 +244,43 @@ module Pretty =
     open Starling.Core.TypeSystem.Pretty
     open Starling.Core.Var.Pretty
 
+    /// <summary>
+    ///     Hidden building-blocks for AST pretty-printers.
+    /// </summary>
+    module private Helpers =
+        /// Pretty-prints blocks with the given indent level (in spaces).
+        let printBlock (pCmd : 'Cmd -> Doc) (c : 'Cmd list) : Doc =
+            braced (ivsep (List.map (pCmd >> Indent) c))
+
+        /// <summary>
+        ///     Pretty-prints an if-then-else.
+        /// </summary>
+        /// <param name="pCond">Pretty-printer for conditionals.</param>
+        /// <param name="pLeg">Pretty-printer for 'then'/'else' legs.</param>
+        /// <param name="cond">The conditional to print.</param>
+        /// <param name="thenLeg">The 'then' leg to print.</param>
+        /// <param name="elseLeg">The optional 'else' leg to print.</param>
+        /// <typeparam name="Cond">Type of conditionals.</typeparam>
+        /// <typeparam name="Leg">Type of 'then'/'else' leg items.</typeparam>
+        /// <returns>
+        ///     A <see cref="Doc"/> capturing the if-then-else form.
+        /// </returns>
+        let printITE
+          (pCond : 'Cond -> Doc)
+          (pLeg : 'Leg -> Doc)
+          (cond : 'Cond)
+          (thenLeg : 'Leg list)
+          (elseLeg : ('Leg list) option)
+          : Doc =
+            syntaxStr "if"
+            <+> parened (pCond cond)
+            <+> printBlock pLeg thenLeg
+            <+> (maybe
+                    Nop
+                    (fun e -> syntaxStr "else" <+> printBlock pLeg e)
+                    elseLeg)
```

Comment:
Abstracted this out because it accounts both for atomic and non-atomic conditionals.